app/controllers/admin_controller.rb

```diff
@@ -16,25 +16,30 @@ def error_reports
   end
 
   def privileges
-    @privileges = Privilege.all.user_sort({ term: params[:sort], default: :threshold },
-                                          rep: :threshold, name: :name)
-                           .paginate(page: params[:page], per_page: 20)
+    @abilities = Ability.all
   end
 
   def show_privilege
-    @privilege = Privilege.find_by name: params[:name]
+    @ability = Ability.find_by internal_id: params[:name]
     respond_to do |format|
-      format.json { render json: @privilege }
+      format.json { render json: @ability }
     end
   end
 
   def update_privilege
-    @privilege = Privilege.find_by name: params[:name]
-    pre = @privilege.threshold
-    @privilege.update(threshold: params[:threshold]) &&
-      AuditLog.admin_audit(event_type: 'privilege_threshold_update', related: @privilege, user: current_user,
-                           comment: "from <<#{pre}>>\nto <<#{params[:threshold]}>>")
-    render json: { status: 'OK', privilege: @privilege }, status: 202
+    @ability = Ability.find_by internal_id: params[:name]
+    if params[:type] == 'post'
+      pre = @ability.post_score_threshold
+      @ability.update(post_score_threshold: params[:threshold])
+    elsif params[:type] == 'edit'
+      pre = @ability.edit_score_threshold
+      @ability.update(edit_score_threshold: params[:threshold])
+    elsif params[:type] == 'flag'
+      pre = @ability.flag_score_threshold
+      @ability.update(flag_score_threshold: params[:threshold])
+    end && AuditLog.admin_audit(event_type: 'ability_threshold_update', related: @ability, user: current_user,
+                           comment: "#{params[:type]} score\nfrom <<#{pre}>>\nto <<#{params[:threshold]}>>")
+    render json: { status: 'OK', privilege: @ability }, status: 202
```

Comment:
Has Rubocop not complained about this yet? You've got some repeated code in the conditionals that could factor out, and some misalignment in the audit log. I'd suggest simplifying this method to something like:

```suggestion
    @ability = Ability.find_by internal_id: params[:name]
    type = ['post', 'edit', 'flag'].include?(params[:type]) ? params[:type] : nil
    return not_found if type.nil?
    
    pre = @ability.send("#{type}_score_threshold".to_sym)
    @ability.update("#{type}_score_threshold" => params[:threshold])
    AuditLog.admin_audit(event_type: 'ability_threshold_update', related: @ability, user: current_user,
                         comment: "#{params[:type]} score\nfrom <<#{pre}>>\nto <<#{params[:threshold]}>>")
    render json: { status: 'OK', privilege: @ability }, status: 202
```